metrics.go

```diff
@@ -22,20 +32,16 @@ func Timer(key string) *timer {
 }
 
 func (t *timer) Emit() {
-	if !enabled {
+	if !publishEnabled {
 		return
 	}
-
-	go func(key string, start, end time.Time) {
-		select {
-		case statQueue <- NewTimer(
-			fmt.Sprintf("%s.%s", metricPrefix, key),
-			start,
-			end,
-		):
-		default:
-		}
-	}(t.key, t.start, time.Now())
+	timer := NewTimer(
+		fmt.Sprintf("%s.%s", metricPrefix, t.key),
+		t.start,
+		time.Now(),
+	)
+	publishWG.Add(1)
```

Comment:
If you were to run:
```go
kv.Emit()
Shutdown()
```
Publish is run asynchronously. If we moved `Wg.Add` to publish there would be a small chance that the `Wg.Wait` in shutdown executes before the `Wg.Add` in publish and the metric is lost.